http-api/src/lib.rs

```diff
@@ -406,10 +452,7 @@ fn project_info(urn: Urn, paths: Paths) -> Result<Info, Error> {
         git::Browser::new_with_namespace(
             &repo,
             &namespace,
-            git::Branch::remote(
-                &format!("heads/{}", meta.default_branch),
-                &remote.default_encoding(),
-            ),
+            get_remote_head_branch(&meta.default_branch, &meta.delegates[0]).unwrap(),
```

Comment:
You should keep the error handling with `MissingDelegations` by calling `meta.delegates.first()` and if that returns `None`, return `MissingDelegations` using `Option::ok_or`.